frontend/src/pages/projects/screens/detail/data-connections/__tests__/utils.spec.ts

```diff
@@ -0,0 +1,230 @@
+import { KnownLabels } from '~/k8sTypes';
+import { DataConnection, DataConnectionType } from '~/pages/projects/types';
+import {
+  convertAWSSecretData,
+  deleteDataConnection,
+  getDataConnectionDescription,
+  getDataConnectionDisplayName,
+  getDataConnectionId,
+  getDataConnectionProvider,
+  getDataConnectionResourceName,
+  getDataConnectionType,
+  isDataConnectionAWS,
+  isSecretAWSSecretKind,
+} from '~/pages/projects/screens/detail/data-connections/utils';
+import { AWS_KEYS } from '~/pages/projects/dataConnections/const';
+import { deleteSecret } from '~/api';
+import { mockSecretK8sResource } from '~/__mocks__/mockSecretK8sResource';
+
+import { genUID } from '~/__mocks__/mockUtils';
+import { mock200Status, mock404Error } from '~/__mocks__/mockK8sStatus';
+
+jest.mock('~/api', () => ({
+  deleteSecret: jest.fn(),
+}));
+
+const deletesecretMock = deleteSecret as jest.Mock;
+
+const secret = mockSecretK8sResource({});
+const dataConnection: DataConnection = {
+  type: 0,
+  data: {
+    kind: 'Secret',
+    apiVersion: 'v1',
+    metadata: {
+      name: 'test-secret',
+      namespace: 'test-project',
+      uid: genUID('secret'),
+      resourceVersion: '5985371',
+      creationTimestamp: '2023-03-22T16:18:56Z',
+      labels: {
+        [KnownLabels.DASHBOARD_RESOURCE]: 'true',
+        [KnownLabels.DATA_CONNECTION_AWS]: 'true',
+      },
+      annotations: {
+        'opendatahub.io/connection-type': 's3',
+        'openshift.io/display-name': 'Test Secret',
+      },
+    },
+    data: {
+      [AWS_KEYS.NAME]: 'test-secret',
+      AWS_ACCESS_KEY_ID: 'test',
+      AWS_DEFAULT_REGION: 'region',
+      AWS_S3_BUCKET: 'bucket',
+      AWS_S3_ENDPOINT: 'endpoint',
+      AWS_SECRET_ACCESS_KEY: 'test',
+    },
+    type: 'Opaque',
+  },
+};
+
+const createDataConnection = (type: DataConnectionType) => ({
+  ...dataConnection,
+  type,
+});
+
+describe('isSecretAWSSecretKind', () => {
+  it('should return false when secret is not AWSSecretKind', () => {
+    secret.metadata.labels = {};
```

Comment:
Get in the habit of creating your mocks inline or cloning. We don't want a change to another part of the data to be done in another test that causes this one to fail. It's pattern that is prone to error in the future.
```suggestion
    const secret = mockSecretK8sResource({});
    secret.metadata.labels = {};
```